pkg/nuctl/test/deploy_test.go

```diff
@@ -120,6 +120,49 @@ func (suite *DeployTestSuite) TestDeployWithMetadata() {
 	suite.Require().Contains(suite.outputBuffer.String(), "0099887766")
 }
 
+func (suite *DeployTestSuite) TestDeployFailsOnMissingPath() {
+	imageName := fmt.Sprintf("nuclio/deploy-test-%s", xid.New().String())
+
+	err := suite.ExecuteNutcl([]string{"deploy", "reverser", "--verbose", "--no-pull"},
+		map[string]string{
+			"nuclio-src-dir": suite.GetNuclioSourceDir(),
+			"image":          imageName,
+			"runtime":        "golang",
+			"handler":        "main:Reverse",
+		})
+
+	suite.Require().Error(err, "Function code must be provided either in the path or inline in a spec file; alternatively, an image or handler may be provided")
+}
+
+func (suite *DeployTestSuite) TestDeployFailsOnShellMissingPathAndHandler() {
+	imageName := fmt.Sprintf("nuclio/deploy-test-%s", xid.New().String())
+
+	err := suite.ExecuteNutcl([]string{"deploy", "reverser", "--verbose", "--no-pull"},
+		map[string]string{
+			"nuclio-src-dir": suite.GetNuclioSourceDir(),
+			"image":          imageName,
+			"runtime":        "shell",
+		})
+
+	suite.Require().Error(err, "Function code must be provided either in the path or inline in a spec file; alternatively, an image or handler may be provided")
+}
+
+func (suite *DeployTestSuite) TestDeployShellViaHandler() {
+	imageName := fmt.Sprintf("nuclio/deploy-test-%s", xid.New().String())
+
+	err := suite.ExecuteNutcl([]string{"deploy", "", "--verbose", "--no-pull"},
+		map[string]string{
+			"build-command": "apk --update --no-cache add imagemagick",
+			"runtime-attrs": "{\"arguments\": \"- -resize 50% fd:1\"}",
+			"nuclio-src-dir": suite.GetNuclioSourceDir(),
+			"image":          imageName,
+			"runtime":        "shell",
+			"handler":        "convert",
+		})
+
+	suite.Require().NoError(err)
```

Comment:
I think a more proper test should then try to invoke this function and see that it works. Perhaps a different handler would be more appropriate here - or just find `convert` functionality whose result can be more easily tested than image processing - maybe something to do with image metadata where we can expect a known textually-represented result. @pavius if you have any ideas here do chime in

Reply:
You can set the handler to `rev`. This will invoke the `rev` binary which will read `stdin` and return the reverse. The `shell` processor sets the event's body to `stdin` so whatever body you send should be received reversed in the response.

This way you can actually check the deploy worked.